hwmv2: soc: nxp: port kinetis family to hardware model v2

danieldegrasse opened the pull request:
Ports Kinetis SOC family and boards to HWMv2. *Note- this PR depends on #68777 for the definition of `SOC_PART_NUMBER`*

danieldegrasse committed:
soc: nxp: kinetis: convert kinetis SOC family to hardware model V2

Convert kinetis SOC family to hardware model V2. Rework SOC Kconfig and
defconfig definitions to align with hardware model V2 guidelines.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>
Co-authored-by: Emilio Benavente <emilio.benavente@nxp.com>
Co-authored-by: David Leach <david.leach@nxp.com>
Co-authored-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
Co-authored-by: Declan Snyder <declan.snyder@nxp.com>
Co-authored-by: Yves Vandervennet <yves.vandervennet@nxp.com>

Commit:
boards: nxp: port frdm_k22f to hwmv2

Port frdm_k22f to hardware model v2

Signed-off-by: David Leach <david.leach@nxp.com>

danieldegrasse committed:
boards: nxp: port frdm_k64f to hwmv2

Port frdm_k642 to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

mmahadevan108 committed:
boards: nxp: frdm_k82f: port to hwmv2

Port frdm_k82f to hardware model v2

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Commit:
boards: nxp: convert frdm_kl25z to hwmv2

Convert frdm_kl25z to hardware model v2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse committed:
boards: nxp: frdm_kw41z: convert to hwmv2

Convert frdm_kw41z to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert hexiwear to hwmv2

Convert hexiwear to hardware model v2. As multiple SOCs exist on this
board, the hexiwear_k64 and hexiwear_kw40z boards are combined into one
target.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert rddrone_fmuk66 to hwmv2

Convert rddrone_fmuk66 to to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

Commit:
boards: nxp: twr_ke18f: convert to hwmv2

Convert twr_ke18f to hardware model v2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: nxp: convert twr_kv58f220m to hwmv2

Convert twr_kv58f220m to hardware model v2

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse committed:
boards: nxp: convert usb_kw24d512 to hwmv2

Convert usb_kw24d512 to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert ip_k66f to hwmv2

Convert ip_k66f to hardware model v2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

nordicjm commented on soc/nxp/kinetis/Kconfig.soc:
```diff
@@ -0,0 +1,11 @@
+	bool
+
+config SOC_FAMILY
+	string
```
```suggestion
```

nordicjm commented on soc/nxp/kinetis/Kconfig.defconfig:
```diff
@@ -8,4 +10,6 @@ config ADC_INIT_PRIORITY
-source "soc/soc_legacy/arm/nxp_kinetis/*/Kconfig.defconfig.series"
+rsource "*/Kconfig.defconfig"
+
+endif # SOC_FAMILY_KINETIS
```
Seems empty lines are missing from these files

nordicjm commented on soc/nxp/kinetis/k2x/Kconfig.defconfig:
```diff
@@ -0,0 +1,29 @@
+
+if SOC_SERIES_KINETIS_K2X
+
+config SOC_SERIES
+	default "k2x"
```
this goes in the Kconfig.soc file

nordicjm commented on soc/nxp/kinetis/kl2x/Kconfig.defconfig:
```diff
@@ -1,13 +1,14 @@
 
 if SOC_SERIES_KINETIS_KL2X
 
 config SOC_SERIES
 	default "kl2x"
```
Kconfig.soc

nordicjm commented on soc/nxp/kinetis/kwx/Kconfig.defconfig:
```diff
@@ -0,0 +1,37 @@
+
+if SOC_SERIES_KINETIS_KWX
+
+config SOC_SERIES
+	default "kwx"
```
Kconfig.soc

nordicjm commented on boards/nxp/frdm_k22f/Kconfig.frdm_k22f:
```diff
@@ -0,0 +1,8 @@
+config BOARD_FRDM_K22F
+	select SOC_MK22F51212
+	
+config SOC_PART_NUMBER
+	default "MK22FN512VLH12"
```
This should be set in Kconfig.soc

nordicjm commented on boards/nxp/frdm_k64f/Kconfig.frdm_k64f:
```diff
@@ -0,0 +1,7 @@
+config BOARD_FRDM_K64F
+	select SOC_MK64F12
+
+config SOC_PART_NUMBER
+	default "MK64FN1M0VLL12"
```
As above

nordicjm commented on boards/nxp/frdm_k82f/Kconfig.frdm_k82f:
```diff
@@ -0,0 +1,8 @@
+config BOARD_FRDM_K82F
+	select SOC_MK82F25615
+
+config SOC_PART_NUMBER
+	default "MK82FN256VLL15"
```
As above

nordicjm commented on boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z:
```diff
@@ -0,0 +1,8 @@
+config BOARD_FRDM_KL25Z
+	select SOC_MKL25Z4
+
+config SOC_PART_NUMBER
+	default "MKL25Z128VLK4"
```
As above

nordicjm commented on boards/nxp/frdm_kw41z/Kconfig.frdm_kw41z:
```diff
@@ -0,0 +1,8 @@
+config BOARD_FRDM_KW41Z
+	select SOC_MKW41Z4
+
+config SOC_PART_NUMBER
+	default "MKW41Z512VHT4"
```
As above

nordicjm commented on boards/nxp/hexiwear/Kconfig.hexiwear:
```diff
@@ -0,0 +1,15 @@
+	bool
+	select SOC_MKW40Z4
+
+config SOC_PART_NUMBER
+	default "MK64FN1M0VDC12" if BOARD_HEXIWEAR_MK64F12
+	default "MKW40Z160VHT4" if BOARD_HEXIWEAR_MKW40Z4
```
As above

nordicjm commented on boards/nxp/hexiwear/doc/index.rst:
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/hexiwear_kw40z/hexiwear_kw40z_defconfig``
```
Path needs updating

nordicjm commented on boards/nxp/hexiwear/doc/index.rst:
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_kw40z
+   :goals: flash
+
+The Segger RTT console is only available during a debug session. Use ``attach``
+to start one:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_kw40z
```
Board names need updating

nordicjm commented on boards/nxp/hexiwear/doc/index.rst:
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
+.. code-block:: console
+
+   ***** Booting Zephyr OS v1.14.0-rc1 *****
+   Hello World! hexiwear_kw40z
```
Board name will not have soc in it anymore

nordicjm commented on boards/nxp/hexiwear/doc/index.rst:
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_kw40z
```
As above

nordicjm commented on boards/nxp/hexiwear/doc/index.rst:
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
+.. code-block:: console
+
+   ***** Booting Zephyr OS v1.14.0-rc1 *****
+   Hello World! hexiwear_kw40z
```
As above

nordicjm commented on boards/nxp/rddrone_fmuk66/Kconfig.rddrone_fmuk66:
```diff
@@ -2,6 +2,7 @@
-	select SOC_PART_NUMBER_MK66FN2M0VLQ18
+	select SOC_MK66F18
+
+config SOC_PART_NUMBER
```
As above

nordicjm commented on boards/nxp/twr_ke18f/Kconfig.twr_ke18f:
```diff
@@ -0,0 +1,8 @@
+config BOARD_TWR_KE18F
+	select SOC_MKE18F16
+
+config SOC_PART_NUMBER
+	default "MKE18F512VLL16"
```
As above

nordicjm commented on boards/nxp/twr_kv58f220m/Kconfig.twr_kv58f220m:
```diff
@@ -0,0 +1,8 @@
+config BOARD_TWR_KV58F220M
+	select SOC_MKV58F24
+
+config SOC_PART_NUMBER
+	default "MKV58F1M0VLQ24"
```
As above

nordicjm commented on boards/nxp/usb_kw24d512/Kconfig.usb_kw24d512:
```diff
@@ -0,0 +1,8 @@
+config BOARD_USB_KW24D512
+	select SOC_MKW24D5
+
+config SOC_PART_NUMBER
+	default "MKW24D512VHA5"
```
As above

nordicjm commented on boards/segger/ip_k66f/Kconfig.ip_k66f:
```diff
@@ -2,6 +2,7 @@
-	select SOC_PART_NUMBER_MK66FN2M0VMD18
+	select SOC_MK66F18
+
+config SOC_PART_NUMBER
+	default "MK66FN2M0VMD18"
```
As above

iuliana-prodan commented on soc/nxp/kinetis/Kconfig.soc:
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2016 Intel Corporation
```
This seems like a new file. If this is the case the copyright should be updated.

iuliana-prodan commented on soc/nxp/kinetis/k6x/Kconfig.defconfig:
```diff
@@ -1,5 +1,6 @@
 
+# Copyright 2024 NXP
 # Copyright (c) 2014-2016 Wind River Systems, Inc.
 # SPDX-License-Identifier: Apache-2.0
```
Based on [this](https://github.com/zephyrproject-rtos/zephyr/pull/68778/commits/6d7b5706751f7367c1b14eab82d2a442bffa5b1d#r1483946471) comment, the SOC_SERIES from this defconfig should be moved to Kconfig. soc

LE: the same for k8x, ke1xf, kv5x

iuliana-prodan commented on boards/nxp/frdm_kw41z/Kconfig.frdm_kw41z:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2017, NXP
```
copyright update

iuliana-prodan commented on boards/nxp/twr_ke18f/Kconfig.twr_ke18f:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019 Vestas Wind Systems A/S
+# SPDX-License-Identifier: Apache-2.0
```
copyright update?

decsny commented:
@nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?

nordicjm commented:
> @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?

The soc's Kconfig.soc file

decsny commented:
> > @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?
> 
> The soc's Kconfig.soc file

so if I'm understanding you, are you saying we should create a few dozen SOC_PART_NUMBER_XXXXXX kconfigs for the same board file to select anyways?

nordicjm commented:
> > > @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?
> > 
> > 
> > The soc's Kconfig.soc file
> 
> so if I'm understanding you, are you saying we should create a few dozen SOC_PART_NUMBER_XXXXXX kconfigs for the same board file to select anyways?

See https://github.com/zephyrproject-rtos/zephyr/pull/68559/files for an example

decsny commented:
> > > > @nordicjm when you say to set SOC_PART_NUMBER in Kconfig.soc, which Kconfig.soc are you talking about?
> > > 
> > > 
> > > The soc's Kconfig.soc file
> > 
> > 
> > so if I'm understanding you, are you saying we should create a few dozen SOC_PART_NUMBER_XXXXXX kconfigs for the same board file to select anyways?
> 
> See https://github.com/zephyrproject-rtos/zephyr/pull/68559/files for an example

This just seems like an extraneous intermediate step, what would be the purpose?

danieldegrasse commented:
> This just seems like an extraneous intermediate step, what would be the purpose? I suppose it is more clear what value the board should set?

I'd also like some clarification here. There are a few reasons we have the board selecting the SOC part number within the current port, versus the `config SOC_PART_NUMBER_xxx` scheme we use in v1: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_kinetis/k2x/Kconfig.soc#L51:

- The build error generated is the same (at least in my tests). Given this, I don't see how the `config SOC_PART_NUMBER_xxx` values at the SOC level will make it simpler for a customer to port a new board
- NXP tends to produce a lot of part numbers for a given SOC. See here for example (from [this file](https://raw.githubusercontent.com/zephyrproject-rtos/hal_nxp/master/mcux/mcux-sdk/devices/MIMXRT1062/MIMXRT1062.h)): 
```
Processors:          MIMXRT1062CVJ5A
**                          MIMXRT1062CVJ5B
**                          MIMXRT1062CVL5A
**                          MIMXRT1062CVL5B
**                          MIMXRT1062DVJ6A
**                          MIMXRT1062DVJ6B
**                          MIMXRT1062DVL6A
**                          MIMXRT1062DVL6B
**                          MIMXRT1062DVN6B
**                          MIMXRT1062XVN5B
```
My concerns with encoding the SOC part number as a Kconfig are the following:
  - typos are easy to make, and hard to catch. We can only really validate the Zephyr port on boards we have access to, which often means only one part number value really gets tested. A typo in tree will require the customer to send a patch or make a downstream fork of Zephyr to correct the issue.
  - Allowing customers to set the part number directly in the board does not expose the value as a configurable option within `menuconfig`, but it allows the customer to support whatever part number they have, regardless of if NXP has defined a `CONFIG_SOC_PART_NUMBER_xxx` value for it (which may take a bit, for example if a new silicon revision is released and we are slow to enable the EVK)

tejlmand reviewed (changes requested):
@danieldegrasse @decsny @nordicjm thanks for raising an important discussion.

So far no specific decisions has been taken or raised in relation to SoC part number and hwmv2, so we have just continued with the design of v1. On my part, because I wasn't aware that someone had found existing solution too limited.

I will try to provide some input to allow us to move forward.

Before getting into details, then I want to highlight the importance of consistency, so that if we should change for nxp, then we should also change for other vendors.

> There are a few reasons we have the board selecting the SOC part number within the current port, versus the config SOC_PART_NUMBER_xxx scheme we use in v1

but in #68778 and HWMv1 the board is still selecting the SoC part number, but they just do it in two different ways.
In HWMv1 as `select SOC_PART_NUMBER_MK22FN512VLH12` which turns into:
```
SOC_PART_NUMBER_KINETIS_K2X
    string
    default "MK22FN512VLH12" if SOC_PART_NUMBER_MK22FN512VLH12
```
benefit of this, compared to #68778 is the ci check, as example in HWMv1, a typo like this `select SOC_PART_NUMBER_MK22FN512VLH21` (end digit 2 <--> 1 are swapped)

will result in error like:
```
warning: undefined symbol SOC_PART_NUMBER_MK22FN512VLH21:
- Referenced at boards/arm/frdm_k22f/Kconfig.board:6:
```

but with the proposal:
```
config SOC_PART_NUMBER
	default "MK22FN512VLH12"
```
no warning will ever be raised because `SOC_PART_NUMBER` is a string.
Thus, it very easy for a board maintainer to make an invalid string.

When the Kconfig SoC provides a Kconfig symbol and defines the string, then there is a lower risk of errors in the board Kconfig, as an error in the select will result in a warning.
Of course a typo in the Kconfig soc string is not caught directly, but upstream code goes through more reviews and more eyes are looking at the code, so chances are higher that mistakes are caught early.

Also, just looking upstream at a line like:
```
default "MK22FN512VLH21" if SOC_PART_NUMBER_MK22FN512VLH12
```
is faster to spot that something is mismatching, either the string or the setting.

Whereas
```
default "MK22FN512VLH21"
```
is not showing a double string, and no relation to what could be wrong.

> typos are easy to make, and hard to catch.

and that just got worse with this PR, se explanation above.

> A typo in tree will require the customer to send a patch or make a downstream fork of Zephyr to correct the issue.

And he should not send a patch if a Zephyr in-tree board accidentally writes: `default "MK22FN512VLH21"` ?
To me it seems like just pushing mistakes to other files, plus increasing the risk because all in-tree boards using same part number now defines the value of the string --> typos are not caught.
Whereas current scheme actually ensures a valid symbol is selected, and then that symbol sets the part number.
Meaning boards always selects a valid symbol (or CI will fail), thus an error in the string only has to be fixed in the SoC Kconfig.
An error in the soc setting name will need to be updated in all boards, but as those will now show up in CI as selecting an invalid soc part number, then those are very easy to find and fix.
An error in the string approach in this PR must be searched and replaced manually, and open PRs will go unnoticed, and as such PRs are not failing CI, then they'll risk not being fixed.

> Allowing customers to set the part number directly in the board does not expose the value as a configurable option within menuconfig, but it allows the customer to support whatever part number they have, regardless of if NXP has defined a CONFIG_SOC_PART_NUMBER_xxx value for it (which may take a bit, for example if a new silicon revision is released and we are slow to enable the EVK)

and this is still possible with existing method, both v1 and v2.

I assume we're talking oot boards, cause a new in-tree board should of course update the known SoC part numbers.

An oot board can still do:
```
# oot Kconfig.<board> file
config SOC_PART_NUMBER
        default "oot_defined_soc_part_number"
```

or even go for the more full fledged solution:
```
# oot Kconfig.<board> file
config SOC_PART_NUMBER_OOT
        bool

config SOC_PART_NUMBER
        default "oot_defined_soc_part_number" if SOC_PART_NUMBER_OOT

config BOARD_FOO
        select SOC_PART_NUMBER_OOT
```
making easy to remove custom oot soc part definition when an upstream symbol is made available.

Also having a `SOC_PART_NUMBER_x` symbol available makes it possible to test for the part number if need be, compared to string comparison.
And remember, `CONFIG_MISPELLED_SETTING` in code is also caught by ci, as:
```
Found references to undefined Kconfig symbols. If any of these are false
positives, then add them to UNDEF_KCONFIG_WHITELIST in ./scripts/ci/check_compliance.py.
...
CONFIG_SOC_PART_NUMBER_MK22FN512VLH21      <file>:<line>
```

based on above, I'm aligned with @nordicjm that current design should be kept, and I have seen no compelling reasons why the soc part number definition should change.

danieldegrasse commented:
> benefit of this, compared to https://github.com/zephyrproject-rtos/zephyr/pull/68778 is the ci check, as example in HWMv1, a typo like this select SOC_PART_NUMBER_MK22FN512VLH21 (end digit 2 <--> 1 are swapped)

Sure, I was aware of the CI check. Given that this is a benefit for us (IE upstream developers), I'll update this PR to use `select SOC_PART_NUMBER`. However, I also expected that using `select SOC_PART_NUMBER_xxx` with an invalid Kconfig would give the user a useful error (IE by stopping the build at that point). At least in my tests, it does not appear to. Do you know why this is the case? 

When I build [this branch](https://github.com/nxp-zephyr/zephyr/tree/demo/k22f-part-num) for the `frdm_k22f`, I don't see any error related to the undefined `SOC_PART_NUMBER_xxx` Kconfig the board is selecting, but the build fails. If we set an invalid Kconfig in the board's `defconfig` file, then the build does give a warning. I guess my core question here is why does this difference exist, and can we make the build warn (or error out) if a board's Kconfig `select`s an invalid SOC part number?

tejlmand commented:
> However, I also expected that using select SOC_PART_NUMBER_xxx with an invalid Kconfig would give the user a useful error (IE by stopping the build at that point). At least in my tests, it does not appear to. Do you know why this is the case?

Yes, that is because when compliance is run, then `KCONFIG_WARN_UNDEF=y` is set.
But that is default off when running a regular build.

One reason for this is because some Kconfig files / trees are conditionally sourced, such as the `arch` and `soc` trees.
Those trees were only sourced if the arch determined from board folder name matched the arch / soc tree in question.

This further means there could be `select`s in rest of Kconfig tree which would not be available in a particular build, but would be available in compliance where arch and SoC in HWMv1 was sourced using `*` as glob pattern.

Another possibility is use of `selects` into symbols of module's Kconfig files (not possible upstream) but where a module could be removed from manifest.

With the new approach in HWMv2, it could be valuable to see if we can improve the warnings in a normal build, so that undef warnings are also printed, but that is outside the scope of current porting work.

danieldegrasse commented:
> This further means there could be selects in rest of Kconfig tree which would not be available in a particular build, but would be available in compliance where arch and SoC in HWMv1 was sourced using * as glob pattern.

Ok, this makes perfect sense- thanks for clarifying the reasoning. The PR should be updated to follow current guidance with `select SOC_PART_NUMBER_xxx`

dleach02 commented:
@nordicjm @tejlmand, are you good with the changes in response to your feedback? We want to close on this so we can start upstreaming additional soc groups.

@danieldegrasse please rebase your branch

nordicjm commented on soc/nxp/kinetis/ke1xf/Kconfig.soc:
```diff
@@ -1,28 +1,38 @@
-	bool "MKE16F16"
+	bool
+	select SOC_SERIES_KINETIS_KE1XF
+	select HAS_MCUX
 	select HAS_MCUX_FLEXCAN
```
these selections go in `Kconfig`

danieldegrasse replied:
Moved

nordicjm commented on soc/nxp/kinetis/ke1xf/Kconfig.soc:
```diff
@@ -1,28 +1,38 @@
-	bool "MKE18F16"
+	bool
+	select SOC_SERIES_KINETIS_KE1XF
+	select HAS_MCUX
 	select HAS_MCUX_FLEXCAN
```
as above

nordicjm commented on soc/nxp/kinetis/k2x/Kconfig.soc:
```diff
@@ -0,0 +1,41 @@
+	bool
+
+config SOC_PART_NUMBER
+	string
```
```suggestion
```

nordicjm commented on soc/nxp/kinetis/k8x/Kconfig.soc:
```diff
@@ -0,0 +1,43 @@
+	bool
+
+config SOC_PART_NUMBER
+	string
```
```suggestion
```

nordicjm commented on soc/nxp/kinetis/ke1xf/Kconfig.soc:
```diff
@@ -58,7 +68,7 @@ config SOC_PART_NUMBER_MKE18F512VLH16
 
-config SOC_PART_NUMBER_KINETIS_KE1XF
+config SOC_PART_NUMBER
 	string
```
```suggestion
```

nordicjm commented on soc/nxp/kinetis/kl2x/Kconfig.soc:
```diff
@@ -59,7 +53,7 @@ config SOC_PART_NUMBER_MKL25Z64VLK4
 
-config SOC_PART_NUMBER_KINETIS_KL2X
+config SOC_PART_NUMBER
 	string
```
```suggestion
```

nordicjm commented on soc/nxp/kinetis/kv5x/Kconfig.soc:
```diff
@@ -41,7 +47,7 @@ config SOC_PART_NUMBER_MKV58F1M0VLL24
 
-config SOC_PART_NUMBER_KINETIS_KV5X
+config SOC_PART_NUMBER
 	string
```
```suggestion
```

nordicjm commented on soc/nxp/kinetis/kwx/Kconfig.soc:
```diff
@@ -0,0 +1,56 @@
+	bool
+
+config SOC_PART_NUMBER
+	string
```
```suggestion
```

danieldegrasse commented on boards/nxp/twr_kv58f220m/doc/twr_kv58f220m.jpg:
Ok- left the image untouched

danieldegrasse commented on boards/nxp/hexiwear/doc/index.rst:
```diff
@@ -334,3 +334,185 @@ will then see a plot of the heart rate data that updates once per second.
+
+The default configuration can be found in the defconfig file:
+
+	``boards/nxp/hexiwear/hexiwear_mkw40z4_defconfig``
```
Fixed

danieldegrasse commented on soc/nxp/kinetis/soc.yml:
```diff
@@ -0,0 +1,31 @@
+family:
+- name: nxp_kinetis
```
Updated family name to `kinetis` in YAML

manuargue reviewed (commented):
shall we add `boards/nxp/index.rst` containing the toctree?

dleach02 commented:
@nordicjm @tejlmand Daniel has responded the the last round of comments. I don't want to keep pinging on this but given the stated objectives to have this branch ready for merge soon after v3.6 we are using the Kinetis PR as a pipe-cleaner before submitting our other platforms.

dleach02 commented:
> shall we add `boards/nxp/index.rst` containing the toctree?

done

nordicjm commented:
Compliance issue needs fixing
Old board names in tests/samples need updating

danieldegrasse committed:
boards: nxp: add toctree placeholder

Add table of contents tree placeholder for NXP boards, in preparation
for porting boards to HWMv2

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>
